### PR TITLE
Bug 1069476 - improved accessibility for the data usage limit view.

### PR DESCRIPTION
**Pull request by yzen**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1392801ff810)

File: apps/costcontrol/locales/costcontrol.en-US.properties [hunk 1]
```diff
@@ -197,6 +202,13 @@ B = B
 KB = KB
 MB = MB
 GB = GB
+# LOCALIZATION NOTE (unit-*): For switch unit button a label describing the
+# unit. The following strings are spoken by screen readers and not shown on the
+# screen.
+unit-B.ariaLabel = B
```
Comment (gitmai) [inline apps/costcontrol/locales/costcontrol.en-US.properties]: This is not necessary.

File: apps/costcontrol/locales/costcontrol.en-US.properties [hunk 2]
```diff
@@ -197,6 +202,13 @@ B = B
...
+unit-KB.ariaLabel = KB
```
Comment (gitmai) [inline apps/costcontrol/locales/costcontrol.en-US.properties]: This is not necessary

File: apps/costcontrol/test/unit/settings_test.js [hunk 3]
```diff
@@ -698,11 +698,13 @@ suite('Settings Test Suite >', function() {
         dataLimitInput.value = '124.56';
         dataLimitInput.dispatchEvent(evtInput);
 
-        var limitUnitValue = dataLimitSwitchUnitButton
-          .querySelector('span.tag');
-        assert.equal(limitUnitValue.textContent, 'MB');
+        assert.equal(dataLimitSwitchUnitButton.getAttribute('data-l10n-id'),
+          'unit-MB');
```
Comment (gitmai) [inline apps/costcontrol/test/unit/settings_test.js]: nit: in Usage we usually align the function parameters like:
assert.equal(dataLimitSwitchUnitButton.getAttribute('data-l10n-id'),
                     'unit-MB');

File: apps/costcontrol/test/unit/settings_test.js [hunk 4]
```diff
@@ -698,11 +698,13 @@ suite('Settings Test Suite >', function() {
...
+        assert.equal(limitUnitValue.getAttribute('data-l10n-id'), 'MB');
         triggerEvent(dataLimitSwitchUnitButton, 'click');
-        assert.equal(limitUnitValue.textContent, 'GB');
+        assert.equal(dataLimitSwitchUnitButton.getAttribute('data-l10n-id'),
```
Comment (gitmai) [inline apps/costcontrol/test/unit/settings_test.js]: Same here.

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=6bada451ee01)

File: apps/costcontrol/test/unit/settings_test.js [hunk 5]
```diff
@@ -698,11 +698,13 @@ suite('Settings Test Suite >', function() {
         dataLimitInput.value = '124.56';
         dataLimitInput.dispatchEvent(evtInput);
 
-        var limitUnitValue = dataLimitSwitchUnitButton
-          .querySelector('span.tag');
-        assert.equal(limitUnitValue.textContent, 'MB');
+        assert.equal(dataLimitSwitchUnitButton.getAttribute('data-l10n-id'),
+        'unit-MB');
```
Comment (gitmai) [inline apps/costcontrol/test/unit/settings_test.js]: Please, put 'unit-MB' aligned with dataLimitSwitchUnitButton.

File: apps/costcontrol/test/unit/settings_test.js [hunk 6]
```diff
@@ -698,11 +698,13 @@ suite('Settings Test Suite >', function() {
...
+        assert.equal(limitUnitValue.getAttribute('data-l10n-id'), 'MB');
         triggerEvent(dataLimitSwitchUnitButton, 'click');
-        assert.equal(limitUnitValue.textContent, 'GB');
+        assert.equal(dataLimitSwitchUnitButton.getAttribute('data-l10n-id'),
+        'unit-GB');
```
Comment (gitmai) [inline apps/costcontrol/test/unit/settings_test.js]: Same here.

File: apps/costcontrol/test/unit/settings_test.js (hunk 5 above)
Comment (yzen) [inline apps/costcontrol/test/unit/settings_test.js]: Duh, sorry, the previous comment was formatted so the string was aligned with the start of the one above :). Done now

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bc7635fbe9b8)

**Pull request merged**